atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceIntegrationTest.java

```diff
@@ -328,6 +338,86 @@ public void testCleanCassLocksStateCli() throws Exception {
+        keyValueService.createTable(tableReference, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        keyValueService.addGarbageCollectionSentinelValues(tableReference, ImmutableList.of(CELL));
+
+        putDummyValueAtCellAndTimestamp(
+                tableReference,
+                CELL,
+                Value.INVALID_VALUE_TIMESTAMP,
+                STARTING_ATLAS_TIMESTAMP - 5_000_000);
+
+        // The following implies that the sentinel was written with a timestamp greater than 5M, which is reasonable
+        // as the current time in Atlas is 10M.
+        Map<Cell, Value> results = keyValueService.get(tableReference, ImmutableMap.of(CELL, 1L));
+        byte[] contents = results.get(CELL).getContents();
+        assertThat(Arrays.equals(contents, PtBytes.EMPTY_BYTE_ARRAY), is(true));
+    }
+
+    @Test
+    public void deletionTakesPlaceAtRelativelyCurrentTimestamp() throws Exception {
+        TableReference tableReference =
+                TableReference.createFromFullyQualifiedName("test." + RandomStringUtils.randomAlphanumeric(16));
+        keyValueService.createTable(tableReference, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        byte[] data = PtBytes.toBytes("data");
+        byte[] moreData = PtBytes.toBytes("data2");
+
+        keyValueService.putWithTimestamps(tableReference, ImmutableListMultimap.of(CELL, Value.create(data, 8L)));
+        keyValueService.putWithTimestamps(tableReference, ImmutableListMultimap.of(CELL, Value.create(moreData, 88L)));
+        keyValueService.delete(tableReference, ImmutableListMultimap.of(CELL, 8L));
+
+        // We want to verify that the delete wrote a Cassandra tombstone with timestamp > 5M
+        putDummyValueAtCellAndTimestamp(tableReference, CELL, 8L, STARTING_ATLAS_TIMESTAMP - 5_000_000);
+        Map<Cell, Value> results = keyValueService.get(tableReference, ImmutableMap.of(CELL, 8L + 1));
+        assertThat(results.containsKey(CELL), is(false));
+    }
+
+    @Test
+    public void rangeTombstonesWrittenAtRelativelyCurrentTimestamp() throws Exception {
+        TableReference tableReference =
+                TableReference.createFromFullyQualifiedName("test." + RandomStringUtils.randomAlphanumeric(16));
+        keyValueService.createTable(tableReference, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        keyValueService.deleteAllTimestamps(
+                tableReference,
+                ImmutableMap.of(CELL, 1_234_567L),
+                true);
+
+        putDummyValueAtCellAndTimestamp(tableReference, CELL, 1337L, STARTING_ATLAS_TIMESTAMP - 5_000_000);
+        Map<Cell, Value> results = keyValueService.get(tableReference, ImmutableMap.of(CELL, 1337L + 1));
+        assertThat(results.containsKey(CELL), is(false));
+
+        putDummyValueAtCellAndTimestamp(tableReference, CELL, 1_333_337L, STARTING_ATLAS_TIMESTAMP - 5_000_000);
+        Map<Cell, Value> results2 = keyValueService.get(tableReference, ImmutableMap.of(CELL, Long.MAX_VALUE));
```

Comment:
nits: `results2` is a bit of a lazy name, and I'm not sure what you're testing with this part.

Reply:
I've renamed the variable, and split the test into 2 parts.

Part 1 is to check that the range tombstones have reasonably current Cassandra timestamps, as this test name suggests.

Part 2 (which includes this check) was intended to be a check against confusing the Cassandra and Atlas timestamps - in particular, if the Cassandra timestamp was used as the Atlas timestamp to apply the range tombstone for, then this test will fail.